source/portable/NetworkInterface/NXP1060/NetworkInterface.c

```diff
@@ -0,0 +1,797 @@
+                        eEMACState = xEMAC_SetupPHY;
+
+                        FreeRTOS_printf( ( "Link down!" ) );
+
+                        xSendEventStructToIPTask( &xNetworkEventDown, 0U );
+
+                        /* Wait for the driver to finish initialization. */
+                        uint32_t ulNotificationValue;
+
+                        do
+                        {
+                            ulNotificationValue = ulTaskNotifyTake( pdTRUE, portMAX_DELAY );
+                        } while( !( ulNotificationValue & DRIVER_READY ) );
+                    }
+                    else
+                    {
+                        /* The link is still up. */
+                        bGlobalLinkStatus = true;
+                    }
+                }
+            } while( bGlobalLinkStatus == false );
+        }
+        else
+        {
+            BaseType_t receiving = pdTRUE;
+
+            /* A packet was received, the link must be up. */
+            bGlobalLinkStatus = true;
+
+            while( receiving == pdTRUE )
+            {
+                uint32_t length;
+                const status_t status = ENET_GetRxFrameSize( &( ethernetifLocal->handle ), &length, 0 );
+
+                switch( status )
+                {
+                    case kStatus_Success: /* there is a frame.  process it */
+
+                        if( length )
+                        {
+                            prvProcessFrame( length );
+                        }
+
+                        break;
+
+                    case kStatus_ENET_RxFrameEmpty: /* Received an empty frame.  Ignore it */
+                        receiving = pdFALSE;
+                        break;
+
+                    case kStatus_ENET_RxFrameError: /* Received an error frame.  Read & drop it */
+                        PRINTF( "RX Receive Error\n" );
+                        ENET_ReadFrame( ethernetifLocal->base, &( ethernetifLocal->handle ), NULL, 0, 0, NULL );
+                        /* Not sure if a trace is required.  The MAC had an error and needed to dump bytes */
+                        break;
+
+                    default:
+                        PRINTF( "RX Receive default\n" );
+                        break;
+                }
+            }
+        }
+    }
+}
+
+/**
+ * @brief Callback for ENET interrupts. We have only enabled the Ethernet receive interrupts
+ * in the case of this driver.
+ */
+static void ethernet_callback( ENET_Type * base,
+                               enet_handle_t * handle,
+                               enet_event_t event,
+                               enet_frame_info_t * frameInfo,
+                               void * userData )
+{
+    BaseType_t needsToYield = pdFALSE;
+
+    ( void ) base;
+    ( void ) handle;
+    ( void ) frameInfo;
+    ( void ) userData;
+
+    switch( event )
+    {
+        case kENET_RxEvent:
+            vTaskNotifyGiveFromISR( receiveTaskHandle, &needsToYield );
```

Comment:
In a zero-copy driver, this would be the best place to pass the incoming packet buffer to the tcp stack.

Reply:
Yes, but currently it is not a zero copy driver and thus we need to invoke a task which can allocate the memory and process the frame.